pkg/kubelet/kubelet.go

```diff
@@ -2795,6 +2800,7 @@ func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, *v1.Pod, v1.PodResizeStatus)
 	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, podCopy); !ok {
 		// Log reason and return. Let the next sync iteration retry the resize
 		klog.V(3).InfoS("Resize cannot be accommodated", "pod", podCopy.Name, "reason", failReason, "message", failMessage)
+		kl.recorder.Eventf(podCopy, v1.EventTypeWarning, events.ResizeDeferred, "Resize cannot be accommodated: %s", failReason)
```

Comment:
What does the actual fail message look like in the case that the pod doesn't fit? We want to make sure it includes which resource doesn't fit, and ideally the actual amounts.

Reply:
Got it, I'll look into the different admission errors in the code

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/lifecycle/interfaces.go#L30-L38


Looking at the code, can I set Reason to a specific type @tallclair , such as type Reason string? This will make it easier to judge. If so, i will open other pr do this first. 
like this: 
```go
type Reason string

const (
	AppArmor = "AppArmor"
	...
)

// PodAdmitResult provides the result of a pod admission decision.
type PodAdmitResult struct {
	// if true, the pod should be admitted.
	Admit bool
	// a brief single-word reason why the pod could not be admitted.
	Reason Reason
	// a brief message explaining why the pod could not be admitted.
	Message string
}
```

Reply:
> What does the actual fail message look like in the case that the pod doesn't fit? We want to make sure it includes which resource doesn't fit, and ideally the actual amounts.

Which access errors only return reason and message? The most intuitive way is to return the reason message to the event. But it seems difficult to judge which access errors are Defer and which are error.